File: 백준/Silver/20291. 파일 정리/파일 정리.py
from collections import defaultdict
ext_dict = defaultdict(int)
for _ in range(int(input())):
    name, ext = input().split(".")
    ext_dict[ext] += 1
    
result = sorted(ext_dict.items(), key=lambda x: x[0])

for name, ext in result:
    print(name, ext)

File: 백준/Silver/20291. 파일 정리/파일 정리.kt
import java.util.*

fun main() {
    val extDict = HashMap<String, Int>()
    val scanner = Scanner(System.`in`)

    val n = scanner.nextInt()
    scanner.nextLine() // Consume the newline character after reading the integer

    for (i in 0 until n) {
        val input = scanner.nextLine()
        val (name, ext) = input.split(".")
        extDict[ext] = extDict.getOrDefault(ext, 0) + 1
    }

    val sortedResult = extDict.entries.sortedBy { it.key }

    for ((name, ext) in sortedResult) {
        println("$name $ext")
    }
}